source/concepts.tex

```diff
@@ -0,0 +1,1249 @@
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+There need be no subsumption relationship between \tcode{Destructible<T>}
+and \tcode{is_\-no\-throw_\-destruct\-ible_v<T>}.
+
+\pnum
+\begin{note}
+Unlike the \oldconcept{Destructible} requirements~(\tref{destructible}), this concept
+forbids destructors that are potentially throwing, even if non-throwing.
+\end{note}
+\end{itemdescr}
+
+\rSec2[concepts.lib.corelang.constructible]{Concept \tcode{Constructible}}
+
+\pnum
+The \tcode{Constructible} concept constrains the initialization of a variable of
+a given type with a particular set of argument types.
+
+\indexlibrary{\idxcode{Constructible}}%
+\begin{itemdecl}
+template <class T, class... Args>
+concept Constructible = Destructible<T> && is_constructible_v<T, Args...>;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+There need be no subsumption relationship between \tcode{Constructible<T, Args...>}
+and \tcode{is_const\-ruct\-ible_v<T, Args...>}.
+\end{itemdescr}
+
+\rSec2[concepts.lib.corelang.defaultconstructible]{Concept \tcode{DefaultConstructible}}
+
+\indexlibrary{\idxcode{DefaultConstructible}}%
+\begin{itemdecl}
+template <class T>
+concept DefaultConstructible = Constructible<T>;
+\end{itemdecl}
+
+\rSec2[concepts.lib.corelang.moveconstructible]{Concept \tcode{MoveConstructible}}
+
+\indexlibrary{\idxcode{MoveConstructible}}%
+\begin{itemdecl}
+template <class T>
+concept MoveConstructible = Constructible<T, T> && ConvertibleTo<T, T>;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+If \tcode{T} is an object type, then let \tcode{rv} be an rvalue of type \tcode{T}
+and \tcode{u2} a distinct object of type \tcode{T} equal to \tcode{rv}.
+\tcode{MoveConstructible<T>} is satisfied only if
+
+\begin{itemize}
+\item After the definition \tcode{T u = rv;}, \tcode{u} is equal to \tcode{u2}.
+
+\item \tcode{T(rv)} is equal to \tcode{u2}.
+
+\item If \tcode{T} is not \tcode{const}, \tcode{rv}'s resulting state is valid
+but unspecified\iref{lib.types.movedfrom}; otherwise, it is unchanged.
+\end{itemize}
+\end{itemdescr}
+
+\rSec2[concepts.lib.corelang.copyconstructible]{Concept \tcode{CopyConstructible}}
+
+\indexlibrary{\idxcode{CopyConstructible}}%
+\begin{itemdecl}
+template <class T>
+concept CopyConstructible = MoveConstructible<T> &&
+  Constructible<T, T&> && ConvertibleTo<T&, T> &&
+  Constructible<T, const T&> && ConvertibleTo<const T&, T> &&
+  Constructible<T, const T> && ConvertibleTo<const T, T>;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+If \tcode{T} is an object type, then let \tcode{v} be an lvalue of type (possibly
+\tcode{const}) \tcode{T} or an rvalue of type \tcode{const T}.
+\tcode{CopyConstructible<T>} is satisfied only if
+
+\begin{itemize}
+\item After the definition \tcode{T u = v;}, \tcode{u} is equal to \tcode{v}.
+
+\item \tcode{T(v)} is equal to \tcode{v}.
+\end{itemize}
+
+\end{itemdescr}
+
+\rSec1[concepts.lib.compare]{Comparison concepts}
+
+\rSec2[concepts.lib.compare.general]{General}
```

Comment:
While ISO says "no hanging paragraphs", I'm not enamoured with single-sentence sections; I think having the essentially non-normative intro in the parent section would work, too. @zygoloid?